api/src/main/java/org/xwiki/apps/googleapps/internal/GoogleAppsAuthServiceImpl.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.apps.googleapps.internal;
+
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.user.api.XWikiUser;
+import com.xpn.xwiki.user.impl.xwiki.XWikiAuthServiceImpl;
+import com.xpn.xwiki.web.XWikiRequest;
+
+import java.net.URLEncoder;
+import java.security.Principal;
+import java.util.regex.Pattern;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.servlet.http.HttpSession;
+
+import org.securityfilter.realm.SimplePrincipal;
+import org.slf4j.Logger;
+import org.xwiki.apps.googleapps.CookieAuthenticationPersistence;
+import org.xwiki.apps.googleapps.GoogleAppsAuthService;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.component.annotation.InstantiationStrategy;
+import org.xwiki.component.descriptor.ComponentInstantiationStrategy;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.container.servlet.filters.SavedRequestManager;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.text.StringUtils;
+
+/**
+ * An authenticator that can include a negotiation with the Google Cloud (e.g. Google Drive) services.
+ * This authenticator is created, configured and maintained by the GoogleAppsScriptService.
+ * @since 3.0
+ * @version $Id$
+ */
+@Component
+@InstantiationStrategy(ComponentInstantiationStrategy.PER_LOOKUP)
+public class GoogleAppsAuthServiceImpl extends XWikiAuthServiceImpl
```

Comment:
AFAIK ``XWikiAuthService`` implementations can't be components ATM. This is most probably why you get 2 instances created: one by the component manager and one by the authentication manager.

Reply:
Why would the auth-manager instantiate this component?
Only the GoogleAppsManagerImpl should do so.

Reply:
Indeed, see may comment above. The issue is actually with the ``GoogleAppsManagerImpl`` which implements multiple roles.